### PR TITLE
Optimize all JPEGs within ~95%+ image similarity (41.6% reduction)

### PR DESCRIPTION
**Pull request by ide**
(If you prefer not to take this commit since it affects so many files and has some subjective decisions explained below, I totally understand.)

# Why
This commit optimizes all of the JPEGs while typically retaining 95% or better similarity, using the SSIM metric. The goal of this commit was to optimize file size without noticeably compromising visual image quality. For many of the images, this results in significant file size savings, sometimes over 50%.

# How:
Since one goal is not to lose noticeable image quality, I programmatically chose the JPEG quality level (`-quality`) based on the similarity (dssim score) between each original image and its re-encoded images. That is, every image was re-encoded with the JPEG quality level that was best for it instead of using the same quality level for all images.

I used a script called cjpeg-dssim, which re-encodes images at different quality levels and computes the dssim score until it finds a good quality level for each image. I edited cjpeg-dssim for lower target dssim scores (lower dssim = higher resemblance to the original image). The values I chose were:
```
DSSIM_LOWER_BOUND="0.007000"
DSSIM_UPPER_BOUND="0.008500"
```

This produced image similarity scores around 0.95 to 0.97 according to imagemagick:
```
magick compare -metric SSIM 02_pcb_front.jpg 02_pcb_front_cjpeg_dssim.jpg /dev/null
```

I re-encoded all images with this command that I just let run in the background:
```
find data -name '*.jpg' | parallel -j4 './cjpeg-dssim mozjpeg {}'
```

To compute the change in file size I ran:
```
git diff | grep '\-size' | cut -d ' ' -f2 | paste -sd+ - | bc
# 1150973098
git diff | grep '+size' | cut -d ' ' -f2 | paste -sd+ - | bc
# 672643335
```

Because literally every JPEG was changed, the next time the site is deployed to S3 it will take a long time and consume 672 MB of bandwidth. Afterwards storage and bandwidth costs should drop meaningfully.

# Test Plan

Visually inspecting a couple of sample images showed that the images are high quality. Since the target dssim scores were so low, I didn't expect there to be human-noticeable changes to the image quality.

GitHub has "swipe" and "onion skin" controls to compare old vs. new images on the code review page. Comparing the overlaid images, I find the image quality is virtually identical.

### CODE REVIEW

**Comment by Gekkio** [comment]
Thanks! :smile: 
I compared a couple of images and couldn't spot of difference, so this is a very useful change. I'll definitely try to set up the same tooling on my machine for handling future submissions.

Can you rebase and force push, since the MGL unit commit is now causing a conflict? I'll merge after that.

**Comment by ide** [comment]
Just force pushed right now.

> try to set up the same tooling on my machine for handling future submissions

I might suggest using just mozjpeg (its version of cjpeg) or jpegoptim with a fixed quality level (like 75) if you don't want to compare dssim values.

I did the dssim-based tuning because this isn't my repository and wanted to err on the side of carefully preserving image quality for such a large PR. But when I ran the dssim-based optimizer I modified it to print out the JPEG quality levels and found that it chose levels usually around 75. The highest I saw was 83 and the lowest was 61, but I didn't thoroughly check all of the levels.

In short, just naively using mozjpeg with `-quality 75` or `-quality 80` will produce excellent results. Probably 80% of the benefit for... 5% of the work?

**Comment by Gekkio** [comment]
Thanks! Merging
